### PR TITLE
Update rocm-llvm-mlir to 5.2.0

### PR DESCRIPTION
**Pull request by linedot**
* Update checksum
* Patch for correctly appending rpath in cmake file
patches #818

### CODE REVIEW

**Comment by tpkessler** [comment]
Should we open an upstream issue / PR on the linker flags in CMakeLists.txt?

**Comment by acxz** [comment]
> Patch for correctly appending rpath in cmake file

Please make an issue outlining the error for the patch and then link this PR to it.

**Comment by linedot** [comment]
> > Patch for correctly appending rpath in cmake file
> 
> Please make an issue outlining the error for the patch and then link this PR to it.

Hi, thanks for guidance. Submitted the Issue here: https://github.com/ROCmSoftwarePlatform/llvm-project-mlir/issues/687

**Comment by acxz** [comment]
Thank you, can you make an upstream PR that would close the upstream issue you made and then have that PR as a source instead of committing the patch to this repo?

**Comment by linedot** [comment]
Alright. How about this?

(PR created here: https://github.com/ROCmSoftwarePlatform/llvm-project-mlir/pull/688 )

File: rocm-llvm-mlir/PKGBUILD [hunk 1]
```diff
@@ -4,18 +4,25 @@
...
 _dirname="$(basename $url)-$(basename ${source[0]} .tar.gz)"
 
+prepare() {
+  cd "$_dirname"
+  patch -Np1 < "$srcdir/llvm-project-mlir-fix-rpath-flags.patch"
+}
+
 build() {
   # -fcf-protection is not supported by HIP, see
   # https://github.com/ROCm-Developer-Tools/HIP/blob/rocm-4.5.x/docs/markdown/clang_options.md
```
Comment (tpkessler) [inline rocm-llvm-mlir/PKGBUILD:28]: Please update this link to https://docs.amd.com/bundle/ROCm-Compiler-Reference-Guide-v5.2/page/Appendix_A.html

**Review by tpkessler — CHANGES_REQUESTED** [review]

File: rocm-llvm-mlir/PKGBUILD (hunk 1 above)
Comment (linedot) [reply rocm-llvm-mlir/PKGBUILD]: done

**Comment by tpkessler** [comment]
LGTM! Thank you very much @linedot.

**Review by tpkessler — APPROVED** [review]

**Pull request merged**